src_testbed/graphics.rs

```diff
@@ -78,14 +78,18 @@ impl GraphicsManager {
         collider: ColliderHandle,
     ) {
         let body = body.unwrap_or(RigidBodyHandle::invalid());
-        if let Some(sns) = self.b2sn.get_mut(&body) {
-            for sn in sns.iter_mut() {
+        if let Some(sns) = self.b2sn.remove(&body) {
+            let mut new_sns = vec![];
+            for sn in sns.into_iter() {
                 if let Some(sn_c) = sn.collider {
                     if sn_c == collider {
                         commands.entity(sn.entity).despawn();
+                    } else {
+                        new_sns.push(sn);
                     }
                 }
             }
+            self.b2sn.insert(body, new_sns);
```

Comment:
```suggestion
        if let Some(mut sns) = self.b2sn.remove(&body) {
            sns = sns
                .into_iter()
                .filter(|sn| {
                    let Some(sn_c) = sn.collider else {
                        return false;
                    };
                    if sn_c == collider {
                        commands.entity(sn.entity).despawn();
                        return false;
                    } else {
                        return true;
                    }
                })
                .collect();
            self.b2sn.insert(body, sns);
        }
```

Small nitpick, I did that just to help me understand the change, not sure if which is better, don´t worry about it I'll merge one or the other after discussion with seb.

Reply:
This can be simplified further with `Vec::retain`:

```rust
        if let Some(sns) = self.b2sn.get_mut(&body) {
            sns.retain(|sn| {
                if let Some(sn_c) = sn.collider {
                    if sn_c == collider {
                        commands.entity(sn.entity).despawn();
                        return false;
                    }
                }

                true
            });
        }
```
this also fixes the case where we the scene node doesn’t have a collider attached, in which case we don’t want this method to remove it from the list.

Reply:
(I applied the change.)